### ISSUE TITLE
Ask for camera and microphone access on macOS 10.14 Mojave

### ISSUE THREAD

**Issue by ShGKme** [nextcloud/talk-desktop#50]
Since macOS 10.14 Mojave it is required to ask media access on Mac.

What to do, if the user refuses?

https://www.electronjs.org/docs/latest/api/system-preferences#systempreferencesaskformediaaccessmediatype-macos

**Comment by Gymnae** [comment]
Happened to me as well, the app did not request permission and I'm not able to use microphones or cameras

**Comment by wolfspyre** [comment]
yeah. seems like the cause is already identified and a PR exists to address from *march*  https://github.com/nextcloud/talk-desktop/pull/80
What needs to happen to get that merged and a new build minted?

**Comment by Gymnae** [comment]
macOS is not in focus until version 1 release, apparently

**Comment by bentuna** [comment]
I can confirm that under macOS 14 Sonoma the issue still exists: no asking for camera or microphone permission and therefore no usage for calls possible.

**Comment by cyr06** [comment]
MacOS Sonoma + Talk Version 0.29.0
(https://github.com/nextcloud-releases/talk-desktop/releases/download/v0.29.0/Nextcloud.Talk-darwin-x64-0.29.0.zip)
I wasn't ask to allow mic and camera so impossible to make a call.

**Comment by ShGKme** [comment]
I will try to check this issue again next week 👀

**Comment by benjamindahl** [comment]
hey people, I've managed to get it to work.
Coming from another problem I realised that Talk-Desktop does not ask for camera and microphone permission.
I used https://github.com/DocSystem/tccutil to grant them with the following command manually: 

Download 
or clone the python-script to your computer. Then run `sudo /usr/bin/python3 ./tccutil.py -p /Applications/Nextcloud\ Talk.app -e --microphone --camera`

![Bildschirmfoto 2024-07-08 um 08 13 45](https://github.com/nextcloud/talk-desktop/assets/2698024/85942f81-9893-48d2-87bd-c61cd53f1b44)

<img width="1111" alt="Bildschirmfoto 2024-07-08 um 08 21 08" src="https://github.com/nextcloud/talk-desktop/assets/2698024/2923bef6-c1c8-457f-8c61-d7b06957a795">

**Comment by benjamindahl** [comment]
Suggestion: title of the bug should be changed to just macOS, because it affects all recent macOS versions

**Comment by crckrmn77** [comment]
```
hey people, I've managed to get it to work. Coming from another problem I realised that Talk-Desktop does not ask for camera and microphone permission. I used https://github.com/DocSystem/tccutil to grant them with the following command manually:

Download or clone the python-script to your computer. Then run `sudo /usr/bin/python3 ./tccutil.py -p /Applications/Nextcloud\ Talk.app -e --microphone --camera`
```
Can confirm this worked for me on M2 MBP running Sonoma 14.5.

**Comment by wolfspyre** [comment]
That's cool! Really. Neat find @benjamindahl  THANK YOU FOR SHARING. 

THAT BEING SAID, This is **NOT A FIX** but a (USEFUL) hacky workaround. 

The fix is that the application must actually request the privileges it needs....
This is handled for talk by the browser, but as a standalone client, the permissions must be explicitly declared

(I believe the entitlement needs to be baked into the app cert too but I'm not a mac app dev ;) )

**Comment by andrewhotlab** [comment]
Using version 0.39.0 on macOS 11.7.10: it does not ask neither seems to be possible manually grant permission on mic and camera! :'(

**Comment by andrewhotlab** [comment]
The version 1.0 is working for us on macOS 11.7.x.

**Comment by cyr06** [comment]
works fine great job !
